src/Agent.Worker/DiagnosticLogManager.cs

```diff
@@ -105,6 +105,21 @@ public sealed class DiagnosticLogManager : AgentService, IDiagnosticLogManager
                 File.Copy(agentLogFile, destination);
             }
 
+            // Read and add to logs waagent.conf settings on Linux
+            if (PlatformUtil.RunningOnLinux)
+            {
+                executionContext.Debug("Creating waagent file.");
+                string waagentFile = Path.Combine(supportFilesFolder, "waagentConf.txt");
+
+                string configFileName = "waagent.conf";
+                string filePath = WhichUtil.Which(configFileName);
+                if(string.IsNullOrWhiteSpace(filePath)){
+                    filePath = Directory.GetFiles("/etc", configFileName).FirstOrDefault();
+                }
+                string waagentContent = !string.IsNullOrEmpty(filePath) ? ParseWaagentContent(filePath) : "waagent.conf file is not found";
```

Comment:
Nit: waagentContent is not really exact here - "waagent.conf file is not found" is the error message, but not the file content.